traffic_portal/app/src/common/api/DeliveryServiceSslKeysService.js

```diff
@@ -29,63 +29,54 @@ var DeliveryServiceSslKeysService = function($http, $q, locationUtils, messageMo
 		generateSslKeyForm.deliveryservice = deliveryService.xmlId;
 		generateSslKeyForm.key = deliveryService.xmlId;
 
-		var request = $q.defer();
-        $http.post(ENV.api['root'] + "deliveryservices/sslkeys/generate", generateSslKeyForm)
-        .then(
+        return $http.post(ENV.api['root'] + "deliveryservices/sslkeys/generate", generateSslKeyForm).then(
             function(result) {
-            	messageModel.setMessages([ { level: 'success', text: 'SSL Keys generated and updated for ' + deliveryService.xmlId } ], true);
-                request.resolve(result.data.response);
+            	messageModel.setMessages(result.data.response, true);
```

Comment:
The response to this api call is strange therefore you can either:

a. fix the response or 
b. or change this to:

`messageModel.setMessages([ { level: 'success', text: result.data.response } ], true);`